### ISSUE TITLE
Pain point: Not enough documentation

### ISSUE THREAD

**Issue by Aspire1Inspire2** [enthought/pyql#261]
Hi I am a new user to pyql.
After installation and giving it a try for the past week, I found that the library is lacking tutorial and documentation.
Maybe the library is good enough for the developers of this library, but this library is not providing enough guidance for new comers.

The amount of notebook/script examples are limited. And they do not systematically explain how to utilize the library.
As a comparison, the documentation and start guide of pandas library does a good job bringing new users into the door.

This library might be a good one. But the potential is largely limited by the bar it sets for the users. The user has to understand well how Quantlib or SWIG implementation works before to take full advantage of this lib.

Though I prefer pythonic way of handling things, I will have to switch to SWIG implementation for this reason.
Hope this can be improved in the future.

Thank you for your work all the time.

**Comment by mrslezak** [comment]
Yes this is 100% true, the documentation is so outdated; if the maintainers want people to extend the library the readme.md and installation instructions should be updated.  Otherwise, it's an esoteric library with only those that "dive in" and explore the source commits to understand the current status / functionality of the libraries.  I don't care honestly if Enthought wants to create a commercial version that actually includes support; as I'm stuck using QuantLib-SWIG, which has a book released last year, on an inhouse implementation for Shell Trading.  Documentation is key for users to contribute.

**Comment by thrasibule** [comment]
Agreed that the documentation needs improvements. I've refreshed it a little bit, and thanks to github action it gets rebuilt on every merge now. You can find the most up to date version here: http://docs.enthought.com/pyql/
Another source of knowledge are the scripts in the examples folder, as well as the tests, which show how to use a lot of the classes.

**Comment by yttasdfghjk** [comment]
+1
